engine/access/rest/websockets/data_providers/base_provider.go

```diff
@@ -0,0 +1,64 @@
+package data_providers
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+
+	"github.com/onflow/flow-go/engine/access/subscription"
+)
+
+// BaseDataProvider defines the basic interface for a data provider. It provides methods
+// for retrieving the provider's unique ID, topic, and a method to close the provider.
+type BaseDataProvider interface {
+	// ID returns the unique identifier of the data provider.
+	ID() uuid.UUID
+	// Topic returns the topic associated with the data provider.
+	Topic() string
+	// Close terminates the data provider.
+	Close() error
+}
+
+var _ BaseDataProvider = (*BaseDataProviderImpl)(nil)
+
+// BaseDataProviderImpl is the concrete implementation of the BaseDataProvider interface.
+// It holds common objects for the provider.
+type BaseDataProviderImpl struct {
```

Comment:
do you think this will be used outside of the `data_provider` module? If not, it's a good candidate to leave unexported (along with its constructor)

Reply:
[ff3851b](https://github.com/onflow/flow-go/pull/6636/commits/ff3851b346f1c9a5e638dd1f54ae1a5506c8e0bd)